Modules/_sqlite/connection.c

```diff
@@ -836,7 +836,9 @@ static void _pysqlite_drop_unused_cursor_references(pysqlite_Connection* self)
 static callback_context *
 create_callback_context(pysqlite_state *state, PyObject *obj)
 {
+    PyGILState_STATE gstate = PyGILState_Ensure();
     callback_context *ctx = PyMem_Malloc(sizeof(callback_context));
+    PyGILState_Release(gstate);
     if (ctx != NULL) {
         ctx->obj = Py_NewRef(obj);
```

Comment:
`Py_NewRef` is also Python API and should be protected by the GIL.

Reply:
Got it. Thanks!